tests/ui/lint/dangling-ptr/instantly-dangling-pointer-issue123613.rs

```diff
@@ -0,0 +1,13 @@
+#![deny(instantly_dangling_pointer)]
+
+const MAX_PATH: usize = 260;
+fn main() {
+    let str1 = String::with_capacity(MAX_PATH).as_mut_ptr();
+    //~^ ERROR [instantly_dangling_pointer]
```

Comment:
Instead of just checking the lint name in those UI tests, could you instead check the actual primary message, that way we make sure that the message is correct and that the variables (the type here) are correctly displayed.

```suggestion
    //~^ ERROR getting a pointer from a temporary `String` will result in a dangling pointer
```

Reply:
The output is checked anyways in `.stderr` files

Reply:
The output is checked but not enforced. There is nothing enforcing that the type displayed is the one we wanted.

> Although UI tests have a `.stderr` file which contains the entire compiler output, UI tests require that errors are also annotated within the source. This redundancy helps avoid mistakes since the `.stderr` files are usually auto-generated.

From ["UI - tests" in `rustc-dev-guide`](https://rustc-dev-guide.rust-lang.org/tests/ui.html#error-annotations).

Btw, I don't think I have ever seen the syntax `ERROR [lint_name]` used anywhere in the test suite, except for error codes, so let's continue that way.